addons/promote/common/src/main/java/org/commonjava/indy/promote/data/PromotionManager.java

```diff
@@ -682,14 +690,31 @@ private PathsPromoteResult runPathPromotions( final PathsPromoteRequest request,
         return contents;
     }
 
-    private void clearStoreNFC( ArtifactStore store )
+    /**
+     * NOTE: Adding sourcePaths parameter here to cut down on number of paths for clearing from NFC.
+     *
+     * @param sourcePaths The set of paths that need to be cleared from the NFC.
+     * @param store The store whose affected groups should have their NFC entries cleared
+     * @throws IndyDataException
+     */
+    private void clearStoreNFC( final Set<String> sourcePaths, ArtifactStore store )
             throws IndyDataException
     {
+        Set<String> paths = sourcePaths.stream()
+                                       .map( sp -> sp.startsWith( "/" ) && sp.length() > 1 ? sp.substring( 1 ) : sp )
+                                       .collect( Collectors.toSet() );
+
         nfc.clearMissing( LocationUtils.toLocation( store ) );
```

Comment:
Why not use the same approach when cleaning the target store? i.e., 
paths.forEach( path -> nfc.clearMissing(new ConcreteResource(LocationUtils.toLocation(store), path) );

Reply:
I'm not sure I follow. Are you saying use the approach I used here for another piece of code elsewhere? If so, where is that?